robot-server/tests/conftest.py

```diff
@@ -148,6 +150,23 @@ def server_temp_directory() -> Iterator[str]:
     yield new_dir
 
 
+@pytest.fixture()
+def clean_server_state() -> Iterator[None]:
```

Comment:
```suggestion
def clean_server_state(run_server: object) -> Iterator[None]:
```

Depending on the `run_server` fixture ensures that Pytest will wait to tear down `run_server` until after this fixture runs its teardown. Without this, I think there's theoretically a hazard of this fixture running too late and running into "connection refused" errors, depending the order in which Pytest chooses to run fixtures, which I think is undefined when there are no explicit dependency relationships.